src/cargo/util/toml_mut/upgrade.rs

```diff
@@ -77,7 +80,10 @@ fn set_comparator(
 fn assign_partial_req(
     version: &semver::Version,
     mut pred: semver::Comparator,
-) -> semver::Comparator {
+) -> Option<semver::Comparator> {
+    if !pred.pre.is_empty() {
```

Comment:
Better to have common for the meaning of `Option` here or on `set_comparator`.

Reply:
That's a good idea. I moved the changes up to the `upgrade_requirement` to avoid fixing them here